src/AzureExtension/DevBox/Helpers/WingetConfigWrapper.cs

```diff
@@ -182,56 +207,61 @@ private void SetStateForCustomizationTask(TaskJSONToCSClasses.BaseClass response
-                        Thread.Sleep(TimeSpan.FromSeconds(30));
-                        break;
-                    }
-                    else if (_oldUnitState[i] != unitState)
-                    {
+                    // individual task statuses.
+                    if (responseStatus == "WaitingForUserSession")
+                    {
+                        isWaitingForUserSession = true;
+                        continue;
+                    }
+
+                    if (responseStatus == "Running")
+                    {
+                        isAnyTaskRunning = true;
+                    }
+
+                    if (_oldUnitState[i] != responseStatus)
+                    {
                         var task = _units[i];
-                        ConfigurationSetStateChangedEventArgs args = new(new(ConfigurationSetChangeEventType.UnitStateChanged, setState, unitState, null, task));
+                        var unitState = DevBoxOperationHelper.JSONStatusToUnitStatus(responseStatus);
+                        var resultInfo = responseStatus == "Failed" ? _commonfailureResult : null;
+                        ConfigurationSetStateChangedEventArgs args = new(new(ConfigurationSetChangeEventType.UnitStateChanged, setState, unitState, resultInfo, task));
                         ConfigurationSetStateChanged?.Invoke(this, args);
-                        _oldUnitState[i] = unitState;
+                        _oldUnitState[i] = responseStatus;
                     }
-
-                    _log.Information($"Unit Status: {unitState}");
+                }
+
+                // If waiting for user session and no task is running, show the adaptive card
+                // We add a wait since Dev Boxes take a little over 2 minutes to start applying
+                // the configuration and we don't want to show the same message immediately after.
+                if (isWaitingForUserSession && !isAnyTaskRunning)
+                {
+                    ApplyConfigurationActionRequiredEventArgs eventArgs = new(new WaitingForUserAdaptiveCardSession(_resumeEvent));
+                    ActionRequired?.Invoke(this, eventArgs);
+                    WaitHandle.WaitAny(new[] { _resumeEvent });
+                    Thread.Sleep(TimeSpan.FromSeconds(135));
                 }
 
                 break;
 
-            case "Succeeded":
-                List<ApplyConfigurationUnitResult> unitResults = new();
-                for (var i = 0; i < _units.Count; i++)
-                {
-                    var task = _units[i];
-                    unitResults.Add(new(task, ConfigurationUnitState.Completed, false, false, null));
-                }
-
-                _applyConfigurationSetResult = new(null, unitResults);
-                break;
-
             case "ValidationFailed":
                 _openConfigurationSetResult = new(new FormatException(Resources.GetResource(ValidationFailedKey)), null, null, 0, 0);
```

Comment:
Do we log this somewhere?

Reply:
No, we don't log this since there isn't any info sent back from the REST API. We just show this on the UI.